qiskit/circuit/library/arithmetic/polynomial_pauli_rotations.py

```diff
@@ -246,7 +246,7 @@ def _reset_registers(self, num_state_qubits):
             # set new register of appropriate size
             qr_state = QuantumRegister(num_state_qubits, name='state')
             qr_target = QuantumRegister(1, name='target')
-            num_ancillas = min(self.degree - 1, num_state_qubits - 2)
+            num_ancillas = 0
```

Comment:
Since we're setting this to 0 we can simplify the logic below a bit, right?

Reply:
changed in the code